README.md

```diff
@@ -1,95 +1,201 @@
+
+- Normalize/Sanitize/Validate an object
+- For more detailed documentation about awesomize, visit the [awesomize documentation](https://github.com/influentialpublishers/awesomize)
+
+Awesomize has four components, all of which are optional: **read -> sanitize -> validate -> normalize**
+
+The **`read`** component has access to the entire object passed to the awesomize function. Here, we'll use Ramda to target a specific value of the object passed:
+
+```
+awesomize: (v) => ({
+  id : {
+    read : R.path([ 'order', 'id' ])
+
+...
+```
+
+**`sanitize`** is an awesomize component that can manipulate the data before it is validated.
 
-## Example Usage
+```
+awesomize: (v) => ({
+  id : {
+    read : R.path([ 'order', 'id' ])
+  , sanitize : [ R.drop(2) ]
 
+...
 ```
-const router     = require('express').Router()
-const JW         = require('jigawatt')
 
-// MIDDLEWARE
-const Order      = require('../middleware/order.js')
-const Customer   = require('../middleare/customer.js')
-const Product    = require('../middleware/product.js')
-const Shipping   = require('../middleare/shipping.js')
+**`validate`** is a validator that is passed to our function as `v`.  Awesomize's `validate` component has a few built-in validator methods such as `required`, `isInt`, `isFunction`, etc...
 
+```
+awesomize: (v) => ({
+  id : {
+    read : R.path([ 'order', 'id' ])
+  , validate : [ v.required, v.isInt ]
 
-// ROUTES
-router.get('order/:orderId', JW(Order.getById));
+...
+```
 
-router.get('order/:orderId/detail', JW(
+We can also chain validation methods, as seen above. For more info on awesomize validators, visit the [documentation](https://github.com/influentialpublishers/awesomize#built-in-validators)
 
-  [ Order.getById             // call all of these promise
-  , Customer.getByOrderId     // functions at the same time
-  , Product.getByOrderId      // and merge request.data when
-  , Shipping.getByOrderId     // all are complete
-  ]
+**`normalize`** is the last awesomize component, called after the data has been validated.
 
-, Order.uniteDetails          // transform-only middleware
-                              // to aggregate details and
-                              // present to user
-))
 ```
+awesomize: (v) => ({
+  id : {
+    read      : R.path([ 'order', 'id' ])
+  , validate  : [ v.required, v.isInt ]
+  , normalize : [ R.inc ]
+  }
+})
+```
+
+A complete awesomize function can be used to awesomize more than one value:
+
+```
+awesomize: (v) => ({
+  id : {
+    read      : R.path([ 'order', 'id' ])
+  , validate  : [ v.required, v.isInt ]
+  }
+, product  : { sanitize : [ R.trim ] }
+, customer : { validate : [ v.required ] }
+})
+```
+
+
+#### io
+**`io :: Request -> Data -> Object a`**
+
+- Can merge new data into object
+- Returns a promise
+
+```
+io: (req, data) => ({
+    orderId  : data.id
+  , product  : data.product
+  , customer : data.customer
+  , quantity : db.getQuantityByOrderId(data.id) // Grab value from database
```

Comment:
This is better called a dao (data access object) vs. a db object. Alternatively you could call it a model, but DAO fits here :)